### ISSUE TITLE
F.clip type restriction is too strict

### ISSUE THREAD

**Issue by niboshi** [chainer/chainer#5812]
Revision: Current master 8e0d84ef53322029524ce24ca457fe1f51ff9e70
Code:
```
x = chainer.Variable(numpy.array([7], numpy.float32))
# Both of the following do not work
# y = F.clip(x, numpy.float32(-1), numpy.float32(1))
# y = F.clip(x, -1, 1)
```
Result:
```
TypeError: x_min must be float value
```

**Comment by grafi-tt** [comment]
Probably it's better to update `F.huber_loss` backward too.

* https://github.com/chainer/chainer/blob/v6.0.0b1/chainer/functions/loss/huber_loss.py#L51

**Comment by niboshi** [comment]
@grafi-tt 
Thank you for pointing out! Fixed in #5835.
